ivy/functional/backends/numpy/experimental/elementwise.py

```diff
@@ -337,17 +337,29 @@ def zeta(
     *,
     out: Optional[np.ndarray] = None,
 ) -> np.ndarray:
-    inf_indices = np.equal(x, 1)
-    temp = np.logical_and(np.not_equal(x, 1), np.less_equal(q, 0))
-    nan_indices = np.logical_or(temp, np.less(x,1))
+    arr_shape = q.shape
+    x, q = x.flatten(), q.flatten()
+
+    inf_indices1 = np.where(x == 1)
+    inf_indices2 = np.intersect1d(
+        np.array(np.where(q == 0)), np.array(np.where(x > 1))
+    )
+    inf_indices3 = np.intersect1d(
```

Comment:
Also, this intersect1d does not work at all when the given inputs is more than 1d.

Reply:
Wouldn't flattening the input arrays solve that ?

Reply:
Then how would you preserve the structure of the input to the output?

Reply:
especial the case when the two inputs do not have the same shape as each other

Reply:
for tensorflow, the two inputs are allowed to be with same shape or having only one of them of length 1 in dimensions. like:
```
import tensorflow as tf
result1 = tf.math.zeta(tf.convert_to_tensor([[1,2],[3,4]], dtype=tf.float32), tf.convert_to_tensor([3,2], dtype=tf.float32))
print(result1)
```

Reply:
which is another hard feature to implement